src/types/events/voice_gateway/ready.rs

```diff
@@ -7,15 +7,27 @@ use std::net::Ipv4Addr;
 use crate::types::WebSocketEvent;
 use serde::{Deserialize, Serialize};
 
+use super::VoiceEncryptionMode;
+
 #[derive(Debug, Deserialize, Serialize, Clone, PartialEq, Eq)]
-/// The ready event for the webrtc stream;
-/// Used to give info after the identify event;
-/// See https://discord.com/developers/docs/topics/voice-connections#establishing-a-voice-websocket-connection-example-voice-ready-payload;
+/// The voice gateway's ready event;
+///
+/// Gives the user info about the udp connection ip and port, srrc to use,
+/// available encryption modes and other data.
+///
+/// Sent in response to an Identify event.
+///
+/// See <https://discord-userdoccers.vercel.app/topics/voice-connections#ready-structure>
 pub struct VoiceReady {
-    ssrc: i32,
-    ip: Ipv4Addr,
-    port: u32,
-    modes: Vec<String>,
+    /// See <https://developer.mozilla.org/en-US/docs/Web/API/RTCRtpStreamStats/ssrc>
+    pub ssrc: u32,
+    pub ip: Ipv4Addr,
+    pub port: u16,
+    /// The available encryption modes for the udp connection
```

Comment:
<!--
 Copyright (c) 2024 star
 
 This Source Code Form is subject to the terms of the Mozilla Public
 License, v. 2.0. If a copy of the MPL was not distributed with this
 file, You can obtain one at https://mozilla.org/MPL/2.0/.
-->


```suggestion
    /// The available encryption modes for the UDP connection
```